compiler/rustc_ast/src/util/classify.rs

```diff
@@ -40,15 +40,49 @@ pub fn expr_trailing_brace(mut expr: &ast::Expr) -> Option<&ast::Expr> {
             | Range(_, Some(e), _)
             | Ret(Some(e))
             | Unary(_, e)
-            | Yield(Some(e)) => {
+            | Yield(Some(e))
+            | Yeet(Some(e))
+            | Become(e) => {
                 expr = e;
             }
             Closure(closure) => {
                 expr = &closure.body;
             }
             Gen(..) | Block(..) | ForLoop(..) | If(..) | Loop(..) | Match(..) | Struct(..)
-            | TryBlock(..) | While(..) => break Some(expr),
-            _ => break None,
+            | TryBlock(..) | While(..) | ConstBlock(_) => break Some(expr),
+            Break(_, _)
+            | Range(_, _, _)
+            | Ret(_)
+            | Yield(_)
```

Comment:
Can you make this more precise? The only pattern that must be matched here is `Yield(None)`. Same with a few of the other below patterns.